### DISCUSSION TITLE
<title> How do I ensure a raw download directly from Github has the default CRLF line endings instead of LF line endings?

### DISCUSSION THREAD

**Discussion by emaiiii** [community/community#112294] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hello Everybody,

If I attempt to do a raw download of a file directly from GitHub, the file will contain LF line endings when the default file has CRLF line endings. I see that when I clone the repo, Github converts the file to have CRLF endings. Is there potentially a way where I can have the file saved in my GitHub repo with the default CRLF line endings?

**Comment by MostafaWaleed0** [comment]
1. Create a `.gitattributes`
2. Add rules to specify line ending behavior: `* text eol=crlf`
3. Commit the `.gitattributes` file to your repository:
```bash
git add .gitattributes
git commit -m "Configure line endings"
git push origin <branch_name>
```

**Reply by emaiiii** [reply]
I have tried this solution. It appears that the use of a .gitattributes file handles commit and checkout processes, but it doesn't affect raw downloads directly on browser.
